Algorithms/Encoders/NysiisEncoder.cs

```diff
@@ -0,0 +1,212 @@
+using System;
+using System.Text;
+using System.Linq;
+using System.Globalization;
+using System.Collections.Generic;
+
+namespace Algorithms.Encoders
+{
+    /// <summary>
+    /// Class for NYSIIS encoding strings.
+    /// </summary>
+    public class NysiisEncoder
+    {
+        private static char[] Vowels = { 'A', 'E', 'I', 'O', 'U' };
+
+        /// <summary>
+        /// Encodes a string using the NYSIIS Algorithm
+        /// </summary>
+        /// <param name="text">The string to encode</param>
+        /// <returns>The NYSIIS encoded string (all uppercase)</returns>
+        public string Encode(string text)
+        {
+            text = text.ToUpper(CultureInfo.CurrentCulture);
+            text = StartReplace(text, new (string, string)[] {
+                ("MAC", "MCC"), ("KN", "NN"), ("K", "C"),
+                ("PH", "FF"), ("PF", "FF"), ("SCH", "SSS")
+            });
+            text = EndReplace(text, new (string, string)[] {
+                ("EE", "Yb"), ("IE", "Yb"),
+                ("DT", "Db"), ("RT", "Db"), ("NT", "Db"),
+                ("ND", "Db")
+            });
+
+            StringBuilder nysiis = new StringBuilder($"{text[0]}");
+            for(int i = 1; i < text.Length; i++)
+            {
+                if(Vowels.Contains(text[i]))
+                {
+                    //Vowel: EV -> AF, otherwise ? -> A
+                    text = (i < text.Length - 1 && text[i] == 'E' && text[i + 1] == 'V') ?
+                        text.ReplaceAt(i , "AF", 2) :
+                        text.ReplaceAt(i, "A", 1);
+                }
+                text = RepWith(text, i, new (char, char)[] {
+                    ('Q', 'G'), ('Z', 'S'), ('M', 'N')
+                });
+
+                if(text[i] == 'K')
+                {
+                    //KN -> NN, otherwise K -> C
+                    text = text.ReplaceSingle(i,
+                        (i < text.Length - 1 && text[i + 1] == 'N') ? 'N' : 'C');
+                }
+
+                text = RepArr(text, i, new (string, string)[] {
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6311443)

Reply:
Have you tried doing that? It works for me, in my example at least.
```suggestion
                text = RepArr(text, i, new[] {
```
![image](https://user-images.githubusercontent.com/29201949/95680841-04bf6300-0be5-11eb-8eb3-e1c4a444e831.png)